### ISSUE TITLE
Add `ifF` on Functor

### ISSUE THREAD

**Issue by kubukoz** [typelevel/cats#3040]
Signature:

```scala
//trait Functor[F[_]]
def ifF[A](fb: F[Boolean])(ifTrue: A, ifFalse: A): F[A]
```

Similar to `ifM` from Monad, but doesn't flatten.

Will be closed by #3058

**Comment by stremlenye** [comment]
Shouldn't it go to the `IfFunctorOps` like it was done for `Apply` and `FlatMap`?

**Comment by kubukoz** [comment]
I think it could be either of these places. Both in the type class with a default implementation and in the syntax. Probably only in the syntax until we drop 2.11 support though.

**Comment by joroKr21** [comment]
That reads like "if and only if" :smile: 
Should the branches be by-name though?

**Comment by kubukoz** [comment]
For laziness, they probably should, yeah.

Good point about the name, but it matches ifA and ifM.

**Comment by gagandeepkalra** [comment]
shouldn't ifTrue and ifFalse be by name parameters ? 

`def ifF[A](fb: F[Boolean])(ifTrue: => A, ifFalse: => A): F[A] = map(fb)(x => if (x) ifTrue else ifFalse)`

**Comment by vasiliybondarenko** [comment]
Could anybody review my solution?

**Comment by kubukoz** [comment]
Merged to master in #3058!
